### DISCUSSION TITLE
prawn encryption password in section

### DISCUSSION THREAD

**Discussion by pavanscs** [prawnpdf/prawn#1198] [Q&A]
i am facing a problem with prawn PDF encryption. it is working but when it is opened with a password and i have to reload the section to rise the password security request. what i need is in a same section if i close the file and reopen the pdf again it have to ask for password without reloading the page/ section. can any once has the solution for this problem.

**Comment by pointlessone** [comment]
I suspect your PDF reader is trying to be helpful and remembers the password for you.

You can try watching the encrypted file. I'm pretty sure it's not changed once you enter the password. And if that's the case, how it can be rendered if the file is still encrypted?

**Reply by pavanscs** [reply]
thanks for the reply, my problem is i can render the pdf in my mail but once i opened it with a password and closed it. now when i open it again it has to ask for password request right but its not asking bro, and if i reload the page/ server then it is asking. i want a solution for that even i open it again it has to ask for password to open it without reloading the page or leaving the section

**Reply by pointlessone** [reply]
I don't there's a solution in Prawn. Maybe check PDF Reference for this but I don't think you'll find anything that can prevent your reader from remembering the password if it wants to.

Reloading page probably works because it results in a document different in a way that makes your reader think it doesn't know the password to decrypt it.

**Reply by pavanscs** [reply]
okay bro thank you, is there any other solution for this like can you suggest any gem that completes my requirement. (password encryption)

**Reply by pointlessone** [reply]
I don't thing this a PDF library issue. But if you happen to find another gem that solves it for you please let me know.
